### ISSUE TITLE
"gulp test --watch" runs in headless mode

### ISSUE THREAD

**Issue by snapwich** [prebid/Prebid.js#1416]
## Type of issue
bug / regression

## Description
The tests run in chrome headless now (I think in relation to #1355, which is a good change IMO), however it should probably not run headless when using `--watch` because then it becomes hard to debug.

## Steps to reproduce
run `gulp test --watch`

### Expected results
browser opens with access to debug console

### Actual results
no browser opens

**Comment by matthewlane** [comment]
Try `gulp test --watch --browsers=chrome` to get non-headless Chrome. Looks like the README should be updated with this

**Comment by snapwich** [comment]
This could be as easy as updating the README.md if that's the case.  Are there instances where `--watch` would be useful with headless?  If not we may want it to default to non-headless if `--watch` is specified.

Also, I don't know if this is intended but the browser that pops up when using non-headless no longer has any output concerning the test results in the debug window.  In the past it would have a test suite overview page, now it looks like all I get is a dot in the top-left of the page.  The debugger still works, however.

**Comment by dbemiller** [comment]
yeah.. I did that to simplify the logic in the build, because it was actually pretty convoluted before. Lots of branching on `env === TRAVIS`,  `env === CI`, `--browsers=stuff`, etc. It's so much easier to remember "HeadlessChrome unless otherwise specified".

I'm honestly not sure why the test suite overview doesn't show anymore. [This thread](https://github.com/karma-runner/karma/issues/793) called it expected behavior, but obviously ours was showing them before. I'm not sure if that's because our version of karma was even older than this thread (2013), or because the newer versions of our karma reporters don't write it to the page anymore, or because I removed some plugin which was making it work.

I can look into these if it's important to you guys... but the consensus around here was that nobody really cared. Everyone already had aliases for `gulp test --watch` anyway, and didn't care whether or not the test output was visible on the page, as long as it printed to the terminal.

Let me know if you guys feel differently.
